### ISSUE TITLE
Vocabulary term "scheme" not defined when used inside securityDefinitions

### ISSUE THREAD

**Issue by nikosft** [w3c/wot-thing-description#915]
When I try to parse a TD that includes `securityDefinitions` (using for example jsonld.js) I receive an error for every term included in the `securityDefinitions` collection. For example even this small snippet will fail with `Error: The property "scheme" in the input was not defined in the context`:

```
{
  "@context": "https://www.w3.org/2019/wot/td/v1",
  "id": "urn:dev:ops:32473-WoTLamp-1234",
  "title": "MyLampThing",
  "securityDefinitions": {
    "basic_sc": {
      "scheme": "basic",
      "in": "header"
    }
  },
  "security": [
    "basic_sc"
  ]
}
```

My suspicion is that this is happening because in https://www.w3.org/2019/wot/td/v1 all terms related to  `securityDefinitions` are defined in the `security` collection. As a result, the following (odd) snippet is parsed without errors:

```
{
  "@context": "https://www.w3.org/2019/wot/td/v1",
  "id": "urn:dev:ops:32473-WoTLamp-1234",
  "title": "MyLampThing",
  "securityDefinitions": {
    "basic_sc": {}
  },
  "security": [
    "basic_sc",
    {
      "scheme": "basic",
      "in": "header"
    }
  ]
}
```

JSON-LD playground experiences the same behavior. [Here is snippet 1](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%22https%3A%2F%2Fwww.w3.org%2F2019%2Fwot%2Ftd%2Fv1%22%2C%22id%22%3A%22urn%3Adev%3Aops%3A32473-WoTLamp-1234%22%2C%22title%22%3A%22MyLampThing%22%2C%22securityDefinitions%22%3A%7B%22basic_sc%22%3A%7B%22scheme%22%3A%22basic%22%2C%22in%22%3A%22header%22%7D%7D%2C%22security%22%3A%5B%22basic_sc%22%5D%7D) (notice that `scheme` is not parsed) and [here is snippet 2](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%22https%3A%2F%2Fwww.w3.org%2F2019%2Fwot%2Ftd%2Fv1%22%2C%22id%22%3A%22urn%3Adev%3Aops%3A32473-WoTLamp-1234%22%2C%22title%22%3A%22MyLampThing%22%2C%22securityDefinitions%22%3A%7B%22basic_sc%22%3A%7B%7D%7D%2C%22security%22%3A%5B%22basic_sc%22%2C%7B%22scheme%22%3A%22basic%22%2C%22in%22%3A%22header%22%7D%5D%7D)

**Comment by sebastiankb** [comment]
It seems a jsonld.js problem (note: jsonld.js is also used by the JSON-LD playground). Please see also this issue here #905

**Comment by nikosft** [comment]
Not that  @gkellogg's [Ruby implementation](http://rdf.greggkellogg.net/distiller?command=serialize&format=jsonld&output_format=nquads&verbose) also misses the securityDefinitions part of the first snippet. Also repeated here:

```
{
  "@context": "https://www.w3.org/2019/wot/td/v1",
  "id": "urn:dev:ops:32473-WoTLamp-1234",
  "title": "MyLampThing",
  "securityDefinitions": {
    "basic_sc": {
      "scheme": "basic",
      "in": "header"
    }
  },
  "security": [
    "basic_sc"
  ]
}
```

**Comment by nikosft** [comment]
Note that the following snippet is parsed correctly in both JSON-LD playground and @gkellogg's implementation. I have added a `@type` inside `securityDefinition`

```
{
  "@context": "https://www.w3.org/2019/wot/td/v1",
  "id": "urn:dev:ops:32473-WoTLamp-1234",
  "title": "MyLampThing",
  "securityDefinitions": {
    "basic_sc": {
      "@type":"security",
      "scheme": "basic",
      "in": "header"
    }
  },
  "security": [
    "basic_sc"
  ]
}
```

**Comment by sebastiankb** [comment]
Last Friday we discussed this issue and hereby we confirm that there is a mistake in the definition of `security` and `securityDefinition` in the context file. We planing to publish an errata and already collecting [issues](#920) which will be fixed soon. 

The `"@ type": "Security"` seems to be a temporary workaround that hopefully works for you. Sorry for the inconvenience and thank you for letting us know.

**Comment by takuki** [comment]
During the virtual F2F on 2020-06-26 (see [minutes](https://www.w3.org/2020/06/26-wot-minutes.html#item02)), it was pointed out that errors like this should be caught by the test.

**Comment by vcharpenay** [comment]
this issue has potentially been fixed by #1168. Could you please review @nikosft?

**Comment by sebastiankb** [comment]
from today's TD call, decided to close

@nikosft if you think this issue is not solved, please reopen this issue again
